File: src/game/tools/rpc/output/wrap/StringArrKVArrWrap.cs
using System;
using Common.Net.Simple;

namespace Game 
{
	public class StringArrKVArrWrap : IWrapper
	{	
		public StringKVArrWrap[] StringMap; // map<string,string>
		
		public void Decode(Packet pck)
		{	
			StringMap = new StringKVArrWrap[pck.GetShort()];
			for (int i = 0; i < StringMap.Length; i++)
			{
				StringMap[i] = new StringKVArrWrap();
				StringMap[i].Decode(pck);
				
			}
			
		}
		
        public void Encode(Packet pck)
        {	
        	if (StringMap == null) pck.PutShort((short)0); 
        	else
        	{
	        	pck.PutShort((short)StringMap.Length);
	        	for(int i = 0; i < StringMap.Length; i++)
	        	{
	        		StringMap[i].Encode(pck);
					
	        	}
	        }
        	
        }
	}
}

File: src/game/tools/rpc/output/wrap/StringArrKVArrWrap.go
package wraps

import (
	rpc "tipcat.com/common/rpc/simple"
)

type StringArrKVArrWrap struct {
	
	StringMap []*StringKVArrWrap // map<string,string>
	rpc.Wrapper
}

func (w *StringArrKVArrWrap)Decode(pck *rpc.Packet) rpc.Wrapper {
	
	w.StringMap = make([]*StringKVArrWrap, int(pck.PopInt16()))
	for i := 0; i < len(w.StringMap); i++ {
		w.StringMap[i] = new(StringKVArrWrap)
		w.StringMap[i].Decode(pck)
		
	}
	
	return w
}

func (w *StringArrKVArrWrap)Encode(pck *rpc.Packet) {
	
	if w.StringMap == nil {
		pck.PutInt16(0)
	} else {
    	pck.PutInt16(int16(len(w.StringMap)))
    	for i := 0; i < len(w.StringMap); i++ {
    		w.StringMap[i].Encode(pck);
			
    	}
    }
	
}

